src/components/web-form-renderer.vue

```diff
@@ -193,7 +193,8 @@ const fetchSubmissionAttachments = () => {
   const requestUrl = apiPaths.submissionAttachments(form.projectId, form.xmlFormId, props.instanceId);
   return submissionAttachments.request({
     url: requestUrl,
-    alert: false
+    alert: false,
+    responseType: 'blob', // Handle all file types for attachments.
   })
     .then(transformAttachmentResponse)
```

Comment:
Maybe this line is part of what's confusing. I don't think it's actually an attachment response. The response isn't passed directly to Web Forms. I don't see anything using the return value from `fetchSubmissionAttachments()`, so I think this line could probably go away.

Reply:
It looks like it's used in the `editInstanceOptions` computed ref.

Reply:
https://github.com/getodk/central-frontend/blob/6e6ab26ba34a96da9743949d9ec8a5249bda5005/src/components/web-form-renderer.vue#L213-L224

Reply:
Central creates a submission attachment record for each expected attachment (for each attachment "slot"). That's true even if the attachment isn't uploaded or fails to upload. I guess here, the `editInstance` prop that's passed to Web Forms only includes submission attachments that have been successfully uploaded.